dspace-uclouvain/src/main/java/org/dspace/uclouvain/xmlworkflow/actions/UCLouvainThesisReviewAction.java

```diff
@@ -155,13 +170,36 @@ public ActionResult processRejectPage(Context context, XmlWorkflowItem wfi, Http
         return new ActionResult(ActionResult.TYPE.TYPE_PAGE);
     }
 
+    /**
+     * Process the action 'RETURN_TO_SUBMITTER' which can be performed by a manager and will:
+     *  -> Send the item back to the submitter for modifications.
+     *  -> Add a message (given by the manager) into a metadata field of the item.
+     *  -> The message will than be used to inform the submitter of the needed changes.
+     * 
+     * @param context: The current DSpace context.
+     * @param wfi: The workflow item that is being operated.
+     * @param request: The current request object.
+     * @return: An ActionResult object which represents the output of the action.
+     * @throws SQLException
+     * @throws AuthorizeException
+     */
     public ActionResult processReturnToSubmitter(Context context, XmlWorkflowItem wfi, HttpServletRequest request) throws SQLException, AuthorizeException {
-        // Return the item to the submitter && encode the required changes in a specific metadata field.
         try {
-            // TODO: Add a message to the metadata to explain the return.
             context.turnOffAuthorisationSystem();
+            // Send the item back to submission state.
             this.xmlWorkflowService.sendWorkflowItemBackSubmission(context, wfi, context.getCurrentUser(), "","Send back to submitter for modifications");
+            
+            // Get the mandatory reason form the request object
+            String reason = request.getParameter("reason");
+            if (reason == null || reason.isEmpty()) {
+                this.logger.error("Each return to submitter must have a reason to specify the changes needed.");
```

Comment:
Not need to store anything in the log. This is an user error, not a backend error